interpreter/cling/lib/Interpreter/Interpreter.cpp

```diff
@@ -1377,4 +1405,54 @@ namespace cling {
     T.setState(Transaction::kCommitted);
   }
 
+  void Interpreter::CodeComplete(const std::string& line, size_t& cursor,
+                      std::vector<std::string>& displayCompletions) const {
+    //Get the results
+    const char * const argV = "cling";
+    std::string llvmdir = this->getCI()->getHeaderSearchOpts().ResourceDir;
+    std::string extra_part = "/lib/clang/3.9.0";
+    std::string::size_type i = llvmdir.find(extra_part);
+    if (i != std::string::npos)
+      llvmdir.erase(i, extra_part.length());
+
+    cling::Interpreter childInterpreter(*this, 1, &argV, llvmdir.c_str());
+
+    // Create the CodeCompleteConsumer with InterpreterCallbacks
+    // from the parent interpreter and set the consumer for the child
+    // interpreter.
+    auto callbacks = this->getCallbacks();
+    callbacks->CreateCodeCompleteConsumer(&childInterpreter);
+
+    auto codeCompletionCI = childInterpreter.getCI();
+    clang::Sema &childSemaRef = codeCompletionCI->getSema();
+
+    // Ignore diagnostics when we tab complete.
+    // This is because we get redefinition errors due to the import of the decls.
+    clang::IgnoringDiagConsumer* ignoringDiagConsumer =
+                                            new clang::IgnoringDiagConsumer();
+    childSemaRef.getDiagnostics().setClient(ignoringDiagConsumer, true);
+
+    auto ownerDiagConsumer =
+                        this->getCI()->getSema().getDiagnostics().takeClient();
+    auto clientDiagConsumer =
+                        this->getCI()->getSema().getDiagnostics().getClient();
+    this->getCI()->getSema().getDiagnostics().setClient(
+                                                  ignoringDiagConsumer, false);
+    
+    // The child will desirialize decls from *this. We need a transaction RAII.
+    PushTransactionRAII RAII(this);
+
+    // Triger the code completion.
+    childInterpreter.codeComplete(line, cursor);
+    callbacks->GetCompletionResults(&childInterpreter, displayCompletions);
```

Comment:
I find the name `displayCompletions` confusing; I'd expect a boolean here. Maybe `completions`?